lib/usd/translators/lightWriter.cpp

```diff
@@ -13,54 +13,205 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 //
+#include "lightWriter.h"
 #include <mayaUsd/fileio/primWriterRegistry.h>
-#include <mayaUsd/fileio/translators/translatorRfMLight.h>
+#include <mayaUsd/fileio/translators/translatorLight.h>
+#include <mayaUsd/fileio/utils/adaptor.h>
+
+#include <maya/MFnDirectionalLight.h>
+#include <maya/MGlobal.h>
+#include <pxr/usd/usdLux/light.h>
+#include <pxr/usd/usdLux/distantLight.h>
 
 PXR_NAMESPACE_OPEN_SCOPE
 
-PXRUSDMAYA_DEFINE_WRITER(PxrAovLight, args, context)
-{
-    return UsdMayaTranslatorRfMLight::Write(args, context);
-}
+/// directionalLight
+PXRUSDMAYA_REGISTER_WRITER(directionalLight, PxrUsdTranslators_DirectionalLightWriter);
+PXRUSDMAYA_REGISTER_ADAPTOR_SCHEMA(directionalLight, UsdLuxDistantLight);
 
-PXRUSDMAYA_DEFINE_WRITER(PxrCylinderLight, args, context)
+PxrUsdTranslators_DirectionalLightWriter::PxrUsdTranslators_DirectionalLightWriter(
+        const MFnDependencyNode& depNodeFn,
+        const SdfPath&           usdPath,
+        UsdMayaWriteJobContext&  jobCtx)  : UsdMayaPrimWriter(depNodeFn, usdPath, jobCtx)
 {
-    return UsdMayaTranslatorRfMLight::Write(args, context);
+	UsdLuxDistantLight distantLight = UsdLuxDistantLight::Define(GetUsdStage(), GetUsdPath());
+    _usdPrim = distantLight.GetPrim();
+    if (!TF_VERIFY(
+            _usdPrim,
+            "Could not get UsdPrim for UsdLuxDistantLight at path '%s'\n",
+            GetUsdPath().GetText())) {
+        return;
+    }
 }
 
-PXRUSDMAYA_DEFINE_WRITER(PxrDiskLight, args, context)
+/* virtual */
+void PxrUsdTranslators_DirectionalLightWriter::Write(const UsdTimeCode& usdTime)
 {
-    return UsdMayaTranslatorRfMLight::Write(args, context);
+	MStatus status;
+    UsdMayaPrimWriter::Write(usdTime);
+	// Since write() above will take care of any animation on the light's
+    // transform, we only want to proceed here if:
+    // - We are at the default time and NO attributes on the shape are animated.
+    //    OR
+    // - We are at a non-default time and some attribute on the shape IS animated.
+    if (usdTime.IsDefault() == _HasAnimCurves()) {
+        return;
+    }
+    UsdLuxDistantLight primSchema(_usdPrim);
+    MFnDirectionalLight lightFn(GetDagPath(), &status);
+    if (!status) {
+        // Do this just to get the print
+        CHECK_MSTATUS(status);
+        return;
+    }
+    
+    // First write the base light attributes
+	UsdMayaTranslatorLight::WriteLightAttrs(usdTime, primSchema, lightFn, _GetSparseValueWriter());
```

Comment:
FYI. You will probably get a clang-format lint error in PF - I see a tab in here.

Reply:
fixed, there were several tabs indeed